docs/_Docs_Overview.md

```diff
@@ -26,55 +26,56 @@ Starting from the layer closes to the user:
+All function declarations live in this layer, which constitutes the majority of the colonyNetwork code. Functions that implement a feature or set of related actions are grouped together into a single contract. There are often several logic contracts representing a single logical entity.
 
-Note that this logic distribution is possible due to the [contract upgrade mechanism](/colonynetwork/docs-the-delegate-proxy-pattern/) we use where essentially all logic contracts for an entity, work with the same underlying `EtherRouter` delegate proxy instance.
+For example, the logic for a colony is distributed across `Colony.sol`, `ColonyFunding.sol`, `ColonyPayment.sol` and `ColonyTask.sol`. Likewise for the ColonyNetwork and ReputationMiningCycle entities.
+
+Note that this logic distribution is possible due to the [contract upgrade mechanism](/colonynetwork/docs-the-delegate-proxy-pattern/), in which all functions are called from the same underlying `EtherRouter` delegate proxy instance, regardless of where they are implemented.
 
 **Access layer**
 
-Access management in the network is handled by a group of contracts that underpin all of the application layers. 
+Access management in the network is handled by a group of contracts that underpin all of the application layers.
 
   * `CommonAuthority.sol`
   * `ColonyAuthority.sol`
   * `ColonyNetworkAuthority.sol`
   * `DomainRoles.sol`
-   
+
 These are based on the `DSRoles` and `DSAuth` implementations from the [dappsys library of contracts](https://github.com/dapphub/dappsys-monolithic).
 For a full list of these contracts, see the "Roles and Authority" point below. Also see [modular permissions section](/colonynetwork/docs-modular-permissions) for design details.
 
 **Data Layer**
 
 Data structures, enums, constants and events are declared in a dedicated `*DataTypes.sol` contract, e.g. `ColonyDataTypes.sol`.
 
-For clarity all storage variables are held separately in a `*Storage.sol` contract, e.g. `ColonyStorage.sol`. Storage variable declaration ordering is crucial to be maintained correctly as that ordering cannot change during contract upgrages, due to the [contract upgrade mechanism](/colonynetwork/docs-the-delegate-proxy-pattern/) we use.
+For clarity, all storage variables are held separately in a `*Storage.sol` contract, e.g. `ColonyStorage.sol`. Storage variable declaration ordering is crucial to be maintained; network upgrades and recovery depend on a consistent and clear storage layout. All variabes are commented with slot numbers to support developers.
```

Comment:
> network upgrades and recovery depend on a consistent and clear storage layout
This is insufficient to convey the full meaning.
"ordering cannot change during contract upgrades" <- this is what it's all about. If you insert a new storage variable amongst existing ones for example, it'll break the upgrade even if the storage layout is clear.